third_party/spirv-cross/spirv_common.hpp

```diff
@@ -241,8 +241,29 @@ static inline void fixup_radix_point(char *str, char radix_point)
 	}
 }
 
+static inline std::string convert_to_smallest_string(float f, char locale_radix_point) {
+    char buf[16]; // e.g.: -0.12345678e-12, 16 bytes needed
+    float r;
+    for (int i = 1; i < 9; i++) {
+        sprintf(buf, "%.*g", i, f);
+        sscanf(buf, "%f", &r);
+        if (r == f) {
+            break;
+        }
+    }
+    fixup_radix_point(buf, locale_radix_point);
+
+    // Ensure that the literal is float.
+    if (!strchr(buf, '.') && !strchr(buf, 'e'))
+        strcat(buf, ".0");
+
+    return { buf };
+}
+
 inline std::string convert_to_string(float t, char locale_radix_point)
 {
+    return convert_to_smallest_string(t, locale_radix_point);
+
```

Comment:
this reads a little strange with the original code below.  Maybe we can add a comment for posterity?